UnicornRender/video/include/unicorn/video/Transform.hpp

```diff
@@ -0,0 +1,202 @@
+     * @brief Returns direction vector
+     *
+     * @attention Don't forget to recalculate @sa UpdateModelMatrix to commit all changes to the object
+     *
+     * @return direction
+     */
+    UNICORN_EXPORT glm::vec3 GetDirection() const;
+
+    /**
+     * @brief Returns right vector
+     *
+     * @attention Don't forget to recalculate @sa UpdateModelMatrix to commit all changes to the object
+     *
+     * @return right vector
+     */
+    UNICORN_EXPORT glm::vec3 GetRight() const;
+
+    /**
+     * @brief Returns up vector
+     *
+     * @attention Don't forget to recalculate @sa UpdateModelMatrix to commit all changes to the object
+     *
+     * @return up vector
+     */
+    UNICORN_EXPORT glm::vec3 GetUp() const;
+
+    /**
+     * @brief Returns world translation
+     *
+     * @attention Don't forget to recalculate @sa UpdateModelMatrix to commit all changes to the object
+     *
+     * @return world translation
+     */
+    UNICORN_EXPORT glm::vec3 GetTranslation() const;
+
+    /**
+     * @brief Returns model matrix
+     *
+     * @attention Don't forget to recalculate @sa UpdateModelMatrix to commit all changes to the object
+     *
+     * @return model matrix
+     */
+    UNICORN_EXPORT glm::mat4 const& GetModelMatrix() const;
+
+    /**
+     * @brief Scales object
+     *
+     * Linearly scales each basis component of the mesh vectors by the given scale factors
+     * Example: {1,1,1} is origin scale, {0.5, 0.5, 2} is x and y diminished twice
+     * and z is twice bigger
+     *
+     * @param[in] scale basis scale factors
+     */
+    UNICORN_EXPORT void Scale(glm::vec3 scale);
+
+    /**
+     * @brief Translates by axes
+     *
+     * @param[in] distance per component distance
+     */
+    UNICORN_EXPORT void TranslateLocal(glm::vec3 distance);
+    /**
+     * @brief Translates by world axes
+     *
+     * @param[in] distance per component distance
+     */
+    UNICORN_EXPORT void TranslateWorld(glm::vec3 distance);
+
+    /**
+     * @brief Rotates by all axes counterclockwise
+     *
+     * @param[in] rotation amount of radians at each axis
+     */
+    UNICORN_EXPORT void Rotate(glm::vec3 rotation);
+
+    /**
+     * @brief Rotates by quaternion
+     *
+     * @param[in] rotation quaternion
+     */
+    UNICORN_EXPORT void Rotate(glm::quat rotation);
+
+    /**
+     * @brief Rotates around axis
+     *
+     * @param[in] angleRadians amount of radians to rotate
+     * @param[in] axis axis of rotation
+     */
+    UNICORN_EXPORT void Rotate(float angleRadians, glm::vec3 axis);
+
+    /**
+     * @brief Sets new orientation quaternion
+     *
+     * @param[in] quat new orientation
+     */
+    UNICORN_EXPORT void SetOrientation(glm::quat quat);
+
+    /**
+     * @brief Constructs orientation from per component rotation
+     *
+     * @param[in] rotation per axis rotation
+     */
+    UNICORN_EXPORT void SetRotation(glm::vec3 rotation);
+
+    /** @brief Recalculates model matrix */
```

Comment:
There is no `model matrix` in the class. This method updates several vectors and calculates `m_transformMatrix`

Please update documentation. Also method name does not make much sense - UpdateTransformMatrix?